packages/windows/src/RPA/Windows/keywords/context.py

```diff
@@ -33,5 +46,58 @@ def logger(self):
         return self.ctx.logger
 
     @property
-    def window(self):
-        return self.ctx.window
+    def current_timeout(self) -> float:
+        # This value can change based on `auto.SetGlobalSearchTimeout(...)` calls.
+        return auto.uiautomation.TIME_OUT_SECOND
+
+    def _window_or_none(
+        self, window: "WindowsElement"  # noqa: F821
+    ) -> Optional["WindowsElement"]:  # noqa: F821
+        if window and window.item:
+            try:
+                window.item.BoundingRectangle
+            except COMError:  # pylint: disable=broad-except
+                # Failure to get the bounding rectangle proves that the window doesn't
+                #  exist anymore.
+                return None
+
+            return window
+
+        return None
+
+    @property
+    def anchor(self) -> Optional["WindowsElement"]:  # noqa: F821
+        return self._window_or_none(self.ctx.anchor_element)
+
+    @property
+    def window(self) -> Optional["WindowsElement"]:  # noqa: F821
+        return self._window_or_none(self.ctx.window_element)
+
+    @contextlib.contextmanager
+    def set_timeout(self, timeout: Optional[float] = None):
+        """Context manager that sets a custom temporary `timeout`."""
+        try:
+            if timeout is not None:
+                auto.SetGlobalSearchTimeout(timeout)
+                self.logger.info("Locator timeout set to: %f", self.current_timeout)
+            yield
+        finally:
+            if timeout is not None:
+                auto.SetGlobalSearchTimeout(self.ctx.global_timeout)
```

Comment:
is there any situation where these context managers would be nested? just wondering if the timeouts should be a stack or something

Reply:
Luckily, no, otherwise we should stack the timeouts.

For now the principle is simple: once a timeout is set for a keyword (and you're under the run of this keyword), no matter how many others keyword/methods we call, the original timeout remains set and no other timeout overrides the currently set one. (so the initial set timeout is inherited in all the other stacked calls)